snuba/querylog/__init__.py

```diff
@@ -99,26 +99,29 @@ def _record_cogs(
     result: Union[QueryResult, QueryException, QueryPlanException],
 ) -> None:
     """
-    Record bytes scanned for Generic Metrics Queries per use case.
+    Record bytes scanned for shared resource Queries per use case.
     """
 
-    if (
-        not isinstance(result, QueryResult)
-        or query_metadata.dataset != "generic_metrics"
-    ):
+    if not isinstance(result, QueryResult):
         return
 
     profile = result.result.get("profile")
     if not profile or (bytes_scanned := profile.get("progress_bytes")) is None:
         return
 
-    if (use_case_id := request.attribution_info.tenant_ids.get("use_case_id")) is None:
-        return
+    app_feature = query_metadata.dataset.replace("_", "")
```

Comment:
would we want to add the referrer to this?

Reply:
I don't believe so, as far as cogs is concerned I think we just need a split of the API usage by dataset (and UseCaseID if applicable).

It wouldn't be hard to modify this to add referrer in future but as far as the requirements of the project go it's just dataset level.

Reply:
Is there a way to preserve the dataset names so that we don't have weird data inconsistency ("genericmetrics" vs. "generic_metrics")? Maybe use a different separator character for the use case?

Reply:
cc @fpacifici 

This may be because other sources recording cogs call it "genericmetrics", but I could be wrong and we can change it to `"generic_metrics:transactions"` or similar pattern.